chromium_src/net/http/transport_security_state.cc

```diff
@@ -21,4 +22,118 @@
 
 #endif
 
+#define BRAVE_TRANSPORT_SECURITY_STATE_DELETE_DYNAMIC_DATA_FOR_HOST \
+  if (enabled_sts_hosts_.DeleteDataInAllPartitions(hashed_host)) {  \
+    deleted = true;                                                 \
+  }
+
 #include "src/net/http/transport_security_state.cc"
+
+#undef BRAVE_TRANSPORT_SECURITY_STATE_DELETE_DYNAMIC_DATA_FOR_HOST
+#undef BRAVE_ENABLE_STATIC_PINS
+
+namespace net {
+
+// Non-anonymous helper to friend with SchemefulSite.
+class HSTSPartitionHashHelper {
+ public:
+  static absl::optional<std::string> GetSchemefulSiteRegistrableDomainOrHost(
+      const SchemefulSite& schemeful_site) {
+    if (!schemeful_site.has_registrable_domain_or_host()) {
+      return absl::nullopt;
+    }
+    return schemeful_site.registrable_domain_or_host();
+  }
+};
+
+namespace {
+
+bool IsTopFrameOriginCryptographic(const IsolationInfo& isolation_info) {
+  return isolation_info.top_frame_origin() &&
+         GURL::SchemeIsCryptographic(
+             isolation_info.top_frame_origin()->scheme());
+}
+
+absl::optional<std::string> GetHSTSPartitionHash(
+    const NetworkIsolationKey& network_isolation_key) {
+  DCHECK(base::FeatureList::IsEnabled(features::kBravePartitionHSTS));
+  // An empty or opaque top frame site cannot be used as a partition key, return
+  // a hash which will be treated as a non-persistable partition.
+  if (network_isolation_key.IsTransient() ||
+      !network_isolation_key.GetTopFrameSite().has_value() ||
+      network_isolation_key.GetTopFrameSite()->opaque()) {
```

Comment:
wondering if this affects IPFS URLs

Reply:
I'll try to use `GetTupleOrPrecursorTupleIfOpaque` here instead of dropping opaque origins. Will also take a look at IPFS URLs. They shouldn't use opaque origins, but I'll check anyway.

Reply:
I'll try to use `GetTupleOrPrecursorTupleIfOpaque` here instead of dropping opaque origins. Will also take a look at IPFS URLs. They shouldn't use opaque origins, but I'll check anyway.

Reply:
`GetTupleOrPrecursorTupleIfOpaque` is now used to support sandboxed frames.